homeassistant/components/github/sensor.py

```diff
@@ -1,293 +1,320 @@
-            self.repository_path, **{"params": {"per_page": 1}}
-        )
 
-    async def _update_issues(self):
-        """Update issues data."""
-        self.issues_response = await self._github.repos.issues.list(
-            self.repository_path
-        )
+class GitHubSensorLatestPullEntity(GitHubSensorLatestBaseEntity):
+    """Defines a GitHub latest pull sensor entity."""
 
-    async def _update_releases(self):
-        """Update releases data."""
-        self.releases_response = await self._github.repos.releases.list(
-            self.repository_path
-        )
+    _coordinator_key: CoordinatorKeyType = "issue"
+    _name: str = "Latest Pull Request"
 
-    async def _update_clones(self):
-        """Update clones data."""
-        self.clones_response = await self._github.repos.traffic.clones(
-            self.repository_path
-        )
+    coordinator: RepositoryIssueDataUpdateCoordinator
 
-    async def _update_views(self):
-        """Update views data."""
-        self.views_response = await self._github.repos.traffic.views(
-            self.repository_path
-        )
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return super().available and len(self.coordinator.data.pulls) != 0
 
-    async def _update_pulls(self):
-        """Update pulls data."""
-        response = await self._github.repos.pulls.list(
-            self.repository_path, **{"params": {"per_page": 100}}
-        )
-        if not response.is_last_page:
-            results = await asyncio.gather(
-                *(
-                    self._github.repos.pulls.list(
-                        self.repository_path,
-                        **{"params": {"per_page": 100, "page": page_number}},
-                    )
-                    for page_number in range(
-                        response.next_page_number, response.last_page_number + 1
-                    )
-                )
-            )
-            for result in results:
-                response.data.extend(result.data)
+    @property
+    def native_value(self) -> StateType:
+        """Return the state of the sensor."""
+        return self.coordinator.data.pulls[0].title
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, str | int | None]:
+        """Return the extra state attributes."""
+        pull = self.coordinator.data.pulls[0]
+        return {
+            "url": pull.html_url,
+            "number": pull.number,
+        }
+
+
+class GitHubSensorLatestCommitEntity(GitHubSensorLatestBaseEntity):
+    """Defines a GitHub latest commit sensor entity."""
 
-        self.pulls_response = response
+    _coordinator_key: CoordinatorKeyType = "commit"
+    _name: str = "Latest Commit"
+
+    coordinator: RepositoryCommitDataUpdateCoordinator
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the state of the sensor."""
+        return self.coordinator.data.commit.message
```

Comment:
Long commit messages (especially those from dependabot) go over the 255 character limit for states.

```suggestion
        return (
            f"{self.coordinator.data.commit.message[:252]}..."
            if len(self.coordinator.data.commit.message) > 255
            else self.coordinator.data.commit.message
        )
```

Might also be worth only using the first line of the commit also

Reply:
This applies to all Latest sensors